Nakama/Source/NakamaCore/NakamaCore.build.cs

```diff
@@ -5,45 +5,55 @@
+            {UnrealTargetPlatform.Linux, Tuple.Create(Path.Combine("linux-x64", "libnakama-sdk.so"), Path.Combine("linux-x64", "libnakama-sdk.so"))},
+        };
 
         if (Target.Platform == UnrealTargetPlatform.Mac)
-		{
-		    var frameworkDir = Path.Combine(ModuleDirectory, "libnakama", "macosx-universal", "nakama-sdk.framework");
-            PublicFrameworks.Add(frameworkDir);
-
-            // Currently headers prefix doesn't match framework name (nakama-cpp != nakama-sdk)
-            // so Clang can't find include path automatically and needs some help
-            PublicIncludePaths.Add(Path.Combine(frameworkDir, "Headers"));
-		}
-		else if (Target.Platform == UnrealTargetPlatform.IOS)
-		{
-		    var frameworkDir = Path.Combine(ModuleDirectory, "libnakama", "ios-arm64", "nakama-sdk.framework");
-            PublicFrameworks.Add(frameworkDir);
-            PublicIncludePaths.Add(Path.Combine(frameworkDir, "Headers"));
-		}
-		else if (Target.Platform == UnrealTargetPlatform.Android)
         {
-	        var androidDir = Path.Combine(ModuleDirectory, "libnakama", "android");
-	        PublicIncludePaths.Add(Path.Combine(androidDir,"include"));
-	        
-	        PublicAdditionalLibraries.Add(Path.Combine(androidDir,"libs","armeabi-v7a","libnakama-sdk.so"));
-	        PublicAdditionalLibraries.Add(Path.Combine(androidDir,"libs","arm64-v8a","libnakama-sdk.so"));
-	        PublicAdditionalLibraries.Add(Path.Combine(androidDir,"libs","x86_64","libnakama-sdk.so"));
-
-	        string relAPLPath = Utils.MakePathRelativeTo(Path.Combine(ModuleDirectory, "Nakama_APL.xml"), Target.RelativeEnginePath);
-	        AdditionalPropertiesForReceipt.Add("AndroidPlugin", relAPLPath);
+			string dylibPath;
+            if (Target.Architecture == UnrealArch.Arm64)
+            {
+                dylibPath = Path.Combine(ModuleDirectory, "libnakama", "macosx-arm64", "libnakama-sdk.dylib");
+            }
+            else if (Target.Architecture == UnrealArch.X64)
+            {
+                dylibPath = Path.Combine(ModuleDirectory, "libnakama", "macosx-x64", "libnakama-sdk.dylib");
+            }
+            else
+            {
+                throw new InvalidOperationException("Unrecognized OSX architecture");
+            }
+
+			PublicDelayLoadDLLs.Add(dylibPath);
+			RuntimeDependencies.Add(dylibPath);
+        }
+        else if (Target.Platform == UnrealTargetPlatform.IOS)
+        {
+            var dylibPath = Path.Combine(ModuleDirectory, "libnakama", "ios-arm64", "libnakama-sdk.dylib");
+			PublicDelayLoadDLLs.Add(dylibPath);
+			RuntimeDependencies.Add(dylibPath);
+        }
+        else if (Target.Platform == UnrealTargetPlatform.Android)
+        {
+            var androidDir = Path.Combine(ModuleDirectory, "libnakama", "android");
+            PublicIncludePaths.Add(Path.Combine(androidDir,"include"));
```

Comment:
should there be an android specific include path? this directory seems to be missing

Reply:
Yep this is unnecessary, removing.